app/core/builder/blueprint.py

```diff
@@ -0,0 +1,189 @@
+
+    @staticmethod
+    def _parse_prg_types(prg_type: str) -> ProgramTypes:
+        """Parse Program type from config.yml
+
+        Args:
+            prg_type (str): Value from 
+
+        Raises:
+            ConfigurationError: _description_
+
+        Returns:
+            _type_: _description_
+        """
+        if prg_type == 'C':
+            return ProgramTypes.C
+        elif prg_type == 'CAL':
+            return ProgramTypes.CAL
+        elif prg_type == 'DD':
+            return ProgramTypes.DD
+        elif prg_type == 'DS':
+            return ProgramTypes.DS
+        elif prg_type== 'ENG':
+            return ProgramTypes.ENG
+        elif prg_type == 'FT':
+            return ProgramTypes.FT
+        elif prg_type == 'LP':
+            return ProgramTypes.LP
+        elif prg_type == 'Q':
+            return ProgramTypes.Q
+        elif prg_type == 'SV':
+            return ProgramTypes.SV
+        else:
+            raise ConfigurationError('Program type', prg_type)   
+
+    @staticmethod
+    def _parse_sites(sites: Union[str, List[str]]) -> FrozenSet[Site]:
+        """Parse Sites in config.yml
+
+        Args:
+            sites (Union[str, List[str]]): Option can be a list of sites or a single one
+
+        Returns:
+            FrozenSet[Site]: a frozen site that contains lucupy Site enums 
+                corresponding to each site.
+        """
+
+        def parse_site_specfic(site: str):
+            if site == 'GS':
+                return Site.GS
+            elif site == 'GN':
+                return Site.GN
+            else:
+                raise ConfigurationError('Missing site', site)
+        if sites == 'ALL_SITES':
+        # In case of ALL_SITES option, return lucupy alias for the set of all Site enums
+            return ALL_SITES 
+
+        if isinstance(sites, list):
+            return frozenset(list(map(parse_site_specfic, sites)))
+        else:
+            # Single site case
+            return frozenset([parse_site_specfic(sites)])
+
+    def __iter__(self):
+        return iter((self.time_slot_length,
+                     self.sites,
+                     self.semesters, 
+                     self.program_types,
+                     self.obs_classes))
+
+class OptimizerBlueprint(Blueprint):
+    """Blueprint for the Collector.
+    This is based on the configuration in config.yml.
+    """
+    
+    def __init__(self, algorithm: str) -> None:
+        self.algorithm = OptimizerBlueprint._parse_optimizer(algorithm)
+    
+    @staticmethod
+    def _parse_optimizer(algorithm_name: str):
+        # TODO: Enums are needed but for now is just Dummy
+        # TODO: When GMax is ready we can expand
+        if algorithm_name in 'Dummy':
+            return DummyOptimizer()
+        else:
+            raise ConfigurationError('Optimizer', config.optimizer.name)
+    def __iter__(self):
+        return iter((self.algorithm))
+
+class Blueprints:
```

Comment:
Now we can have Blueprints in just one place and not scatter in the imports. This could be a dataclass? I opt for a simple class so we don't to instance the class.